### ISSUE TITLE
Some CRUD ops on sub-items redirect browser to main page rather than to parent object that was just modified.

### ISSUE THREAD

**Issue by explody** [digitalocean/netbox#2303]
### Environment
* Python version:  3.6.5
* NetBox version:  2.4-beta1 HEAD

It appears that when performing CRUD operations on a some sub-items, specifically interfaces on devices, upon hitting the final "Create/Delete" button, the user is redirected to the main page rather than back to the device. 

Normal editing seems to have no issue, this seems to only occur on sub-items like interfaces.

### Steps to Reproduce

Add or remove interfaces on a device, make sure to hit "Create" and not "Create and add more".  

### Expected Behavior

That the browser would be redirected to the device just edited.

### Observed Behavior

Browser is redirected to the main NB page.

**Comment by jeremystretch** [comment]
Not able to reproduce this. I get directed back to the device page as expected. Can you provide the _exact_ steps you're taking, beginning with creating a new device?

**Comment by explody** [comment]
Yeah, absolutely, I narrowed it down here as well, my original report was too vague.  I've confirmed it when *bulk deleting* interfaces from a device.  I'm directed to a confirmation page as expected, but upon confirming, I'm directed to the main page.

This does not appear to happen when using the trashcan button,  only when checking boxes and hitting 'Delete'.  

I could swear I hit it other spots as well, but I haven't confirmed any others

<img width="1106" alt="screen shot 2018-08-06 at 1 54 40 pm" src="https://user-images.githubusercontent.com/80896/43732758-7a46162a-9980-11e8-813e-48abfd4d41d2.png">
<img width="897" alt="screen shot 2018-08-06 at 1 54 34 pm" src="https://user-images.githubusercontent.com/80896/43732759-7a65a90e-9980-11e8-98da-130cbbcbfbc7.png">
